File: src/app/models/hospitalisation/chambre.ts
import { Lit } from "./lit";

export interface IChambre {
  id: number;
  nom: string;
  slug: string;
  lits: Lit[]
}

export interface ChambreResponse {
  id: number;
  nom: string;
  slug: string;
  lits: Lit[]
}

export class Chambre {
  id: number;
  nom: string;
  slug: string;
  lits: Lit[];

  constructor(iChambre: IChambre) {
    this.id = iChambre.id;
    this.nom = iChambre.nom;
    this.slug = iChambre.slug;
    this.lits = iChambre.lits;
  }

  static fromResponse(chambre: ChambreResponse): Chambre {
    return new Chambre({
      id: chambre.id,
      nom: chambre.nom,
      slug: chambre.slug,
      lits: chambre.lits
    });
  }
}

File: Hospi/src/main/java/com/dopediatrie/hosman/hospi/entity/Chambre.java
package com.dopediatrie.hosman.hospi.entity;

import com.fasterxml.jackson.annotation.JsonIgnore;
import jakarta.persistence.*;
import lombok.AllArgsConstructor;
import lombok.Builder;
import lombok.Data;
import lombok.NoArgsConstructor;

import java.time.LocalDateTime;
import java.util.List;

@Entity
@Data
@AllArgsConstructor
@NoArgsConstructor
@Builder
public class Chambre {
    @Id
    @GeneratedValue(strategy= GenerationType.AUTO)
    private long id;
    private String nom;
    private String slug;
    private LocalDateTime created_at;
    private LocalDateTime updated_at;
    @OneToMany(mappedBy = "chambre")
    private List<Lit> lits;

    @Override
    public String toString() {
        return "Chambre{" +
                "id=" + id +
                ", nom='" + nom + '\'' +
                ", slug='" + slug + '\'' +
                ", created_at=" + created_at +
                ", updated_at=" + updated_at +
                '}';
    }
}
